timelock-impl/src/test/java/com/palantir/atlasdb/timelock/management/DisabledNamespacesTest.java

```diff
@@ -0,0 +1,156 @@
+import com.palantir.atlasdb.timelock.api.ReenableNamespacesRequest;
+import com.palantir.atlasdb.timelock.api.ReenableNamespacesResponse;
+import com.palantir.atlasdb.timelock.api.SuccessfulDisableNamespacesResponse;
+import com.palantir.atlasdb.timelock.api.UnsuccessfulDisableNamespacesResponse;
+import com.palantir.paxos.SqliteConnections;
+import java.util.UUID;
+import javax.sql.DataSource;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+public class DisabledNamespacesTest {
+    private static final UUID LOCK_ID = new UUID(13, 52);
+    private static final UUID OTHER_LOCK_ID = new UUID(123, 45);
+
+    @Rule
+    public TemporaryFolder tempFolder = new TemporaryFolder();
+
+    private static final Namespace FIRST = Namespace.of("fst");
+    private static final Namespace SECOND = Namespace.of("snd");
+
+    private DisabledNamespaces disabledNamespaces;
+
+    @Before
+    public void setup() {
+        DataSource dataSource = SqliteConnections.getDefaultConfiguredPooledDataSource(
+                tempFolder.getRoot().toPath());
+        disabledNamespaces = DisabledNamespaces.create(dataSource);
+    }
+
+    @Test
+    public void notDisabledByDefault() {
+        assertThat(disabledNamespaces.isDisabled(FIRST)).isFalse();
+        assertThat(disabledNamespaces.isDisabled(SECOND)).isFalse();
+        assertThat(disabledNamespaces.disabledNamespaces()).isEmpty();
+    }
+
+    @Test
+    public void canDisableSingleNamespace() {
+        disabledNamespaces.disable(disableNamespacesRequest(FIRST));
+
+        assertThat(disabledNamespaces.isDisabled(FIRST)).isTrue();
+        assertThat(disabledNamespaces.isDisabled(SECOND)).isFalse();
+        assertThat(disabledNamespaces.disabledNamespaces()).containsExactly(FIRST);
+    }
+
+    @Test
+    public void canDisableMultipleNamespaces() {
+        disabledNamespaces.disable(disableNamespacesRequest(FIRST));
+        disabledNamespaces.disable(disableNamespacesRequest(SECOND));
+
+        assertThat(disabledNamespaces.isDisabled(FIRST)).isTrue();
+        assertThat(disabledNamespaces.isDisabled(SECOND)).isTrue();
+        assertThat(disabledNamespaces.disabledNamespaces()).containsExactlyInAnyOrder(FIRST, SECOND);
+    }
+
+    @Test
+    public void disableFailsIfAlreadyDisabled() {
+        DisableNamespacesResponse firstResponse = disabledNamespaces.disable(disableNamespacesRequest(FIRST));
+        assertThat(firstResponse)
+                .isEqualTo(DisableNamespacesResponse.successful(SuccessfulDisableNamespacesResponse.of(LOCK_ID)));
+
+        DisableNamespacesResponse secondResponse = disabledNamespaces.disable(disableNamespacesRequest(FIRST));
+        assertThat(secondResponse)
+                .isEqualTo(DisableNamespacesResponse.unsuccessful(
+                        UnsuccessfulDisableNamespacesResponse.of(ImmutableSet.of(FIRST))));
+    }
+
+    @Test
+    public void disableFailsIfPartiallyDisabled() {
```

Comment:
This is a nice test, but it is not clear if this is due to getting lucky in set ordering or because the transactionality works. Maybe do it with only FIRST disabled, and then with only SECOND disabled -- still technically not deterministic but should be good enough?

Reply:
yeah fair, if the underlying JDBI stuff processes the inputs in a deterministic order, then having both will make it clear that we hit the intended path (especially if the set used in the test is one of those with a guaranteed iteration order).